### DISCUSSION TITLE
routes-manifest.json issue

### DISCUSSION THREAD

**Discussion by selimovicz** [vercel/vercel#4288] [General] [answered]
Hi guys, I stumbled upon the issue which says:

```
11:18:41.073  
Error: A routes-manifest.json couldn't be found. This could be due to a failure during the build
11:18:41.073  
    at Object.getRoutesManifest (/zeit/1bc6bc026129a8fb/.build-utils/.builder/node_modules/@now/next/dist/index.js:6113:15)
11:18:41.073 
```
When I try to build the app locally it succeeds. While using Vercel it always throws me this error.

**Comment by coetry** [comment]
Hi @selimovicz Can you please share a tiny reproducible example for us?

**Comment by wokalski** [comment]
It seems like it's because if there's next in dependencies, now thinks it's a next project 🤦 
https://github.com/zeit/now/blob/f61b36495c4c1573ede1835645e43aee0b2cd588/packages/now-next/src/utils.ts#L323

**Reply by selimovicz** [reply]
I actually resolved this issue by building my project not using web interface but using terminal.

**Reply by wokalski** [reply]
I build the project using now (using a GitHub action specifically) but I still get that. I could work around it by running `yarn remove next` right before the deployment 🙃

**Reply by paulogdm** [reply]
You can use https://vercel.com/blog/advanced-project-settings to edit the settings if you need to. Drop a message on vercel.com/support for further assistance.

**Reply by wokalski** [reply]
@paulogdm I am using advanced project settings but as long as `next` is present in package.json, Vercel will treat it as a next.js project.

**Comment by styfle** [comment, marked as answer]
This issue has been fixed in PR #5009, thanks!

**Comment by hanselabreu** [comment]
This is happening to me now on a `Create React App` project.

**Reply by styfle** [reply]
It sounds like you have Next.js as a dependency and it was auto-detected instead of CRA?

Try changing the framework in your project settings and deploy again.

https://vercel.com/docs/build-step#build-&-development-settings

**Reply by hanselabreu** [reply]
hey Steven, I do have the `Next.js` dependency in my project, however I do have the framework setup to be `Create React App`. 

![image](https://user-images.githubusercontent.com/27902567/94869116-9fb08400-0412-11eb-9857-d18815ea6004.png)

**Reply by hanselabreu** [reply]
It was actually deploying fine, until I added the `Next.js` dependency.

**Reply by styfle** [reply]
I see. The fix in #5009 is only applied to new projects so it seems like this is an older project? Can you try creating a new project with the same code to see if it works?

**Reply by hanselabreu** [reply]
Just tried it. The problem still persists on a new project.

**Reply by styfle** [reply]
I see the problem! I created PR #5246, thanks!

**Reply by hanselabreu** [reply]
you're the best! thanks :)
